std/assembly/util/sort.ts

```diff
@@ -32,10 +32,11 @@ export function COMPARATOR<T>(): (a: T, b: T) => i32 {
       if (a === b || a === null || b === null) return 0;
       var alen = changetype<string>(a).length;
       var blen = changetype<string>(b).length;
-      if (!alen && !blen) return 0;
+      if ((alen | blen) == 0) return 0;
       if (!alen) return -1;
       if (!blen) return  1;
-      return compareImpl(changetype<string>(a), 0, changetype<string>(b), 0, <usize>min(alen, blen));
+      let res = compareImpl(changetype<string>(a), 0, changetype<string>(b), 0, <usize>min(alen, blen));
+      return res == 0 ? alen - blen : res;
```

Comment:
This is a little less clear than before, but also avoids a bit of parsing and optimizing work. An `if` on a `local.get` instead of a `i32.eq` with a constant.

```suggestion
      return res ? res : alen - blen;
```

Again, a minor nit. Not yet sure how much we should focus on these things considering that stdlib parses with every and compiles with many programs. Hmm...